### PR TITLE
Implemented Circuit Breaker

### PR DESCRIPTION
**Pull request by thisismana**
- Added Circuit breaker in `AbstractService`
- passing `ExecutionContexts` now happes as `@Inject` on service level,
no more passing it as `implicit val`
- Deprecated `Loggable` and removed usages in favor of `play.api.Logger`
- Removed `GuiceModules` in favor of `@ImplementedBy` to avoid
instantiating not-needed services
- clean up dependencies in `build.sbt`

### CODE REVIEW

File: core-client/src/main/scala/de/welt/contentapi/core/client/services/configuration/ServiceConfiguration.scala [hunk 1]
```diff
@@ -3,44 +3,94 @@ package de.welt.contentapi.core.client.services.configuration
...
-  * Reusable Service Configuration object based on the [[play.Configuration]]
+  * Reusable Service Configuration object based on the [[play.api.Configuration]]
+  *
+  * Example config:
+  *
+  * {{{
+  * host: https://www.example.com
+  * endpoint: /item/%s
+  * apiKey: dead-beef-1234567-1234-12345567
+  * circuitBreaker {
```
Comment (harryurban) [inline core-client/src/main/scala/de/welt/contentapi/core/client/services/configuration/ServiceConfiguration.scala]: maybe comment with default values?

File: core-client/src/main/scala/de/welt/contentapi/core/client/services/configuration/ServiceConfiguration.scala [hunk 2]
```diff
@@ -3,44 +3,94 @@ package de.welt.contentapi.core.client.services.configuration
...
+    .getOrElse(CircuitBreakerConstants.DefaultFailures)
+  lazy val callTimeout: FiniteDuration = delegate.getOptional[FiniteDuration]("circuitBreaker.callTimeout")
+    .getOrElse(CircuitBreakerConstants.DefaultCallTimeout)
+  lazy val resetTimeout: FiniteDuration = delegate.getOptional[FiniteDuration]("circuitBreaker.resetTimeout")
+    .getOrElse(CircuitBreakerConstants.DefaultResetTimeout)
+  lazy val exponentialBackoff: FiniteDuration = delegate.getOptional[FiniteDuration]("circuitBreaker.exponentialBackoff")
+    .getOrElse(CircuitBreakerConstants.DefaultExponentialBackoff)
+}
+
+object CircuitBreakerConstants {
```
Comment (harryurban) [inline core-client/src/main/scala/de/welt/contentapi/core/client/services/configuration/ServiceConfiguration.scala]: rename `CircuitBreakerDefaults`

File: core-client/src/main/scala/de/welt/contentapi/core/client/services/configuration/ServiceConfiguration.scala [hunk 3]
```diff
@@ -3,44 +3,94 @@ package de.welt.contentapi.core.client.services.configuration
...
+    // make sure, that either apiKey or BasicAuth is set. Not both.
+    if (c.has("credentials") && c.has("apiKey")) {
+      throw c.reportError("credentials", "Both `apiKey` and basic auth `credentials` were configured. Please remove one of those.")
+    }
+
+    (for {
+      credentials ← c.getOptional[Configuration]("credentials")
+      username ← credentials.getOptional[String]("username")
+      password ← credentials.getOptional[String]("password")
+    } yield Right((username, password)))
```
Comment (harryurban) [inline core-client/src/main/scala/de/welt/contentapi/core/client/services/configuration/ServiceConfiguration.scala]: move last `)`to new line

File: core-client/src/main/scala/de/welt/contentapi/core/client/services/contentapi/AbstractService.scala [hunk 4]
```diff
@@ -48,21 +43,31 @@ trait AbstractService[T] extends Strings with Loggable with Status with HeaderNa
     */
   def jsonValidate: JsLookupResult => JsResult[T]
 
+  lazy val breaker: CircuitBreaker = new CircuitBreaker(
```
Comment (harryurban) [inline core-client/src/main/scala/de/welt/contentapi/core/client/services/contentapi/AbstractService.scala]: make `def config` a `lazy val`

File: core-client/src/main/scala/de/welt/contentapi/core/client/services/contentapi/AbstractService.scala [hunk 5]
```diff
@@ -48,21 +43,31 @@ trait AbstractService[T] extends Strings with Loggable with Status with HeaderNa
...
+    capi.actorSystem.scheduler,
+    maxFailures = config.circuitBreaker.maxFailures,
+    callTimeout = config.circuitBreaker.callTimeout,
+    resetTimeout = config.circuitBreaker.resetTimeout)
+    .onOpen(Logger.warn(s"CircuitBreaker [$serviceName] is now open, and will not close for some time"))
```
Comment (harryurban) [inline core-client/src/main/scala/de/welt/contentapi/core/client/services/contentapi/AbstractService.scala]: datadog metrics?

File: core-client/src/test/scala/de/welt/contentapi/core/client/services/contentapi/AbstractServiceTest.scala [hunk 6]
```diff
@@ -80,13 +74,11 @@ class AbstractServiceTest extends PlaySpec with MockitoSugar with Status {
     ))
   }
 
-  object TestServiceApiKeyOverridesBasicAuthConfig {
+  object TestServiceApiKeyApiKey {
```
Comment (harryurban) [inline core-client/src/test/scala/de/welt/contentapi/core/client/services/contentapi/AbstractServiceTest.scala]: ApiKeyApiKey

File: core-client/src/main/scala/de/welt/contentapi/core/client/services/configuration/ServiceConfiguration.scala (hunk 1 above)
Comment (thisismana) [reply core-client/src/main/scala/de/welt/contentapi/core/client/services/configuration/ServiceConfiguration.scala]: done

**Pull request merged**